tree.go

```diff
@@ -771,7 +870,7 @@ func MergeTrees(subroots []*InternalNode) VerkleNode {
 
 func (n *LeafNode) ToHashedNode() *HashedNode {
 	if n.commitment == nil {
-		panic("nil commitment")
+		n.Commit()
```

Comment:
Remember that we tried as much as possible to do "early" `Commit()` when creating leafs and deserializing nodes?

To make things safe, if somebody asks for the commitment and we don't have that calculated, they're forcing us to calculate.

The way that `go-verkle` is used in `go-ethereum` doesn't seem to indicate this pattern should be the normal case. 

Again, if we can't defer the `Commit()` of leaf nodes to be don as late as possible, we're forcing everything to do CPU heavy work in a single core when leafs are created. We need to delegat as much pending work as possible in the root `Commit()` call running in multiple cores.